be/src/storage/column_predicate.h

```diff
@@ -155,9 +155,18 @@ class ColumnPredicate {
 
     virtual bool support_bloom_filter() const { return false; }
 
+    // return true means this predicate can support ngram bloom filter, don't consider gram number(N)
+    // in ngram_bloom_filter(), if gram number is not equal(only happended in ngram_search right now)
+    // it will return true directly. This design is becasue gram number is hard to get in ScalarColumnIterator
+    // and not all predicate need gram size to determin whether support ngram bloom filter
+    virtual bool support_ngram_bloom_filter() const { return false; }
+
     // Return false to filter out a data page.
     virtual bool bloom_filter(const BloomFilter* bf) const { return true; }
 
+    // Return false to filter out a data page.
+    virtual bool ngram_bloom_filter(const BloomFilter* bf, size_t gram_num) const { return true; }
```

Comment:
return false or true?

Reply:
if return false, this page is filtered by index. In default(other Column Predicate), return true. for ColumnExprPredicate, it's determind by _expr_ctxs[0]->support_ngram_bloom_filter(). Real logic is in VectorizedFunctionCallExpr::ngram_bloom_filter